rudder-web/src/main/scala/com/normation/rudder/web/components/popup/RuleModificationValidationPopup.scala

```diff
@@ -74,61 +76,57 @@ object RuleModificationValidationPopup extends Loggable {
    * - Rules
    * - Enable & disable (for rule), delete, modify (save)
    */
-  private def titles = Map(
-      "enable"  -> "Enable a Rule"
-    , "disable" -> "Disable a Rule"
-    , "delete"  -> "Delete a Rule"
-    , "save"    -> "Update a Rule"
-    , "create"  -> "Create a Rule"
-  )
-  private def titleworkflow = Map(
-      "enable"  -> "enable"
-    , "disable" -> "disable"
-    , "delete"  -> "delete"
-    , "save"    -> "update"
-    , "create"  -> "create"
-  )
-  private def explanationMessages = Map(
-      "enable"  ->
-    <div class="row">
+  private def titles(action: RuleModAction) = action match {
+    case RuleModAction.Enable  => "Enable a Rule"
+    case RuleModAction.Disable => "Disable a Rule"
+    case RuleModAction.Delete  => "Delete a Rule"
+    case RuleModAction.Update  => "Update a Rule"
+    case RuleModAction.Create  => "Create a Rule"
+  }
+
+  private def explanationMessages(action: RuleModAction) = action match {
+    case RuleModAction.Enable =>
```

Comment:
Whole function could be replaced by <pre>`
      <div class="row">
        <h4 class="col-lg-12 col-sm-12 col-xs-12 text-center">
            Are you sure that you want to ${action.name} this Rule?
        </h4>
    </div>`</pre>